test/readme.md

```diff
@@ -1,110 +1,268 @@
+| TypeScript Definition | `/test/types` | The TypeScript definition tests verify the type definitions are correct. | `npm run check:tsd` |
 
-Since it's a flat yaml file, you can run the following to get a sourceable environment file:
 
-```sh
-cat serverless-expansion.yml | sed 's/: /=/g' > serverless.env
-```
+## Running the Tests Locally
 
-Before sourcing `serverless.env`, make some adjustments that are equivalent to what our EVG does:
+The easiest way to get started running the tests locally is to start a standalone server and run all of the tests.
 
-- Change `MONGODB_URI` to be the same as `SINGLE_ATLASPROXY_SERVERLESS_URI`
-- Add `SINGLE_MONGOS_LB_URI` and `MULTI_MONGOS_LB_URI` and set them to `SINGLE_ATLASPROXY_SERVERLESS_URI`
+Start a mongod standalone with our [cluster_setup.sh](tools/cluster_setup.sh) script: `./test/tools/cluster_setup.sh server`.
 
-Lastly, comment out the `source` of `install-dependencies.sh` command in `.evergreen/run-serverless-tests.sh` and you can run that script directly to test serverless instances from your local machine.
+Then run the tests: `npm test`.
 
-### Load Balanced
+>**Note:** the command above will run a subset of the tests that work with the standalone server topology since the tests are being run against a standalone server.
+
+The output will show how many tests passed, failed, and are pending. Tests that we have indicated should be skipped using `.skip()` will appear as pending in the test results.  See [Mocha's documentation][mocha-skip] for more information.
+
+In the following subsections, we'll dig into the details of running the tests.
+
+### Testing Different MongoDB Topologies
+
+As we mentioned earlier, the tests check the topology of the MongoDB server being used and run the tests associated with that topology.  Tests that don't have a matching topology will be skipped.
+
+In the steps above, we started a standalone server: `./test/tools/cluster_setup.sh server`.
+
+You can use the same [cluster_setup.sh](tools/cluster_setup.sh) script to start a replica set or sharded cluster by passing the appropriate option: `./test/tools/cluster_setup.sh replica_set` or
+`./test/tools/cluster_setup.sh sharded_cluster`.  If you are running more than a standalone server, make sure your `ulimit` settings are in accordance with [MongoDB's recommendations][mongodb-ulimit]. Changing the settings on the latest versions of macOS can be tricky. See [this article][macos-ulimt] for tips. (You likely don't need to do the complicated maxproc steps.)
+
+The [cluster_setup.sh](tools/cluster_setup.sh) script automatically stores the files associated with the MongoDB server in the `data` directory, which is stored at the top level of this repository.
+You can delete this directory if you want to ensure you're running a clean configuration. If you delete the directory, the associated database server will be stopped, and you will need to run [cluster_setup.sh](tools/cluster_setup.sh) again.
```

Comment:
I'm actually not sure if this is true - I think if the server is running you wouldn't be allowed to delete the directory as the server would have a handle on those files. I think you would need to stop the server first, then delete the directory.

Reply:
This is taken from one of the comments Daria added to https://jira.mongodb.org/browse/NODE-3753. I've tested it out, and the processes are no longer running after deleting the directory. I'm happy to change it if you don't think people should take this approach.

Reply:
I just verified that as well, so fine to stay as is.